### PR TITLE
Use go-ini to read the current vault profile and update values

### PR DESCRIPTION
**Pull request by nihalgonsalves**
Instead of a string template, this uses [ini](https://ini.unknwon.io/docs/intro/getting_started) to read and write to the actual config file.

This means that it can update files in-place, and leaves old values intact (even, for example, if you add a key to a profile that was previously generated, and then re-run the tool).

BREAKING CHANGES

- `vault` now requires `--vault-config-path`
- `switch-roles` now requires `--output-file`

Other changes:

- Added acceptance tests
- feat: Added `--use-role-name-in-profile` to generate profiles such as `accountName_roleName`
- feat: Added `--keep-custom-config` (default true) to allow cleaning out unrelated profiles by setting it to false
- refactor: Decoupled the role fetching from the output generation, i.e. generators simply accept a list of role strings
- chore: Added `make format`

### CODE REVIEW

**Comment by msiuts** [comment]
Would it make sense to add an option to override the file completely? Sometimes manual changes cause trouble and are hard to spot, in that case generating a fresh config and being able to diff might be helpful. WDYT?

**Comment by nihalgonsalves** [comment]
@msiuts Good idea! I'll configure an option to leave the source profile intact and remove everything else.

File: README.md [hunk 1]
```diff
@@ -42,20 +57,27 @@ role_arn=arn:aws:iam::098765432123:role/role-name-two
...
 
-- `source-profile` Can be used to specify a source profile other than `default`
-- `region` Can be used to specify a region other than the region in your source profile
+```
+--source-profile="default"    The profile that your credentials should come from
+--region=STRING               Override the region configured with your source profile
+--vault-config-path=STRING    Where to load/save the config
+--keep-custom-config          Retains any custom profiles or settings. Set to false to remove everything
+                              except the source profile and generated config
+--use-role-name-in-profile    Append the role name to the profile name
```
Comment (nathan-beneke) [inline README.md:71]:
```suggestion
--keep-custom-config=true     Retains any custom profiles or settings. Set to false to remove everything
                              except the source profile and generated config
--use-role-name-in-profile=false Append the role name to the profile name
```

**Review by nathan-beneke — APPROVED** [review]
Before making a new release I want to change the package structure a bit, and also add a verbose/debug mode now that it doesn't print directly to standard out, but this is a great improvement 👍

**Pull request merged**